daemon/src/request/states.rs

```diff
@@ -80,15 +80,18 @@ pub struct Requested {
 /// Note that the related data isn't in the `Status` variants. The status is
 /// free to be associated with any external data, e.g. using it as a value in a
 /// `HashMap`.
-#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub enum Status {
     /// The status of the related data is: ready to go!
     Available,
     /// The status of the related data is: in progress!
     InProgress,
     /// The status of the related data is: failed :(
-    Failed,
+    Failed {
+        #[allow(missing_docs)]
+        reason: String,
```

Comment:
Would it be better to use `Box<dyn Error>` here too?

Reply:
That would require us to have a custom `Serialize` implementation I think, which to me seemed like it wasn't worth it given that this `reason` field is for diagnostic purposes anyway rather than for creating new errors from.